hopefully this works

Hurleveur opened the pull request:
basically does the extra height think we talked about, but also works with all planets having the same size that's sometimes bigger or smaller depending on if there are different non uninhabited governments present in planets in the system

that could be changed without too much difficulty to showing only non uninhabited but I like it better this way not showing when not needed

note: I have NOT tested this yet because I dont have my main pc rn
also maybe I should do a mix of both and still not show uninhabited when both mix but idk if its worth it

Hurleveur committed:
hopefully this works

Hurleveur commented:
can you make the tests run plz @TomGoodIdea 
I'm also very much open to feedback on it, especially on this early stage

TomGoodIdea commented on source/MapPlanetCard.cpp:
```diff
@@ -256,6 +263,24 @@ void MapPlanetCard::Select(bool select)
+
+void MapPlanetCard::ResetSize()
+{
+	lastGovernmentName.empty();
```
```suggestion
```

TomGoodIdea commented:
Enabled checks for my fork, I hope the next commit will trigger them (sorry, still learning how to use Github).

Building locally, after fixing errors, returned some linker problems with the static variables.

TomGoodIdea commented on source/MapDetailPanel.cpp:
```diff
@@ -191,7 +191,7 @@ bool MapDetailPanel::Scroll(double dx, double dy)
 bool MapDetailPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool isNewPress)
 {
 	const Interface *planetCardInterface = GameData::Interfaces().Get("map planet card");
```
unused
```suggestion
```

TomGoodIdea commented on source/MapPlanetCard.cpp:
```diff
@@ -269,20 +294,17 @@ void MapPlanetCard::Highlight(double availableSpace) const
 
 
-double MapPlanetCard::AvailableBottomSpace() const
+double MapPlanetCard::AvailableBottomSpace()
```
Since this function remains unchanged in the header, I assume it wasn't intentional.
```suggestion
double MapPlanetCard::AvailableBottomSpace() const
```

TomGoodIdea reviewed (commented):
build-time errors

Hurleveur committed:
Update source/MapPlanetCard.cpp

Co-authored-by: TomGoodIdea <108272452+TomGoodIdea@users.noreply.github.com>

Hurleveur committed:
Update source/MapDetailPanel.cpp

Co-authored-by: TomGoodIdea <108272452+TomGoodIdea@users.noreply.github.com>

Hurleveur commented:
I think you need to press a button to allow the tests to run for this one PR

Hurleveur committed:
define static members

Hurleveur committed:
smart compiler ^^'

TomGoodIdea commented:
don't know if you've already tested this, but it always displays the government, there's no extra space, and clicking is broken

Hurleveur commented:
~~works about as well as I would have expected then!~~

Hurleveur committed:
fix uninhabited detection?

Hurleveur commented:
I'll notify you once I have tested this but it could be a few days
~~what do you mean clicking is broken? maybe I can fix it by ear, it's just off a bit right?~~
edit: this ought to work, I hope at least x)

Hurleveur committed:
I'm an idiot alright

the government thingy was prob working already
I just forgot to do this lmao

TomGoodIdea commented:
Actually I think there could be a system where all planets have the same gov, but different from the system one. I know it's very unlikely, but we should support every possible situation.

That being said, I think we need to compare the planet's gov name against system's gov name (passed as an argument, or using `planet->GetSystem()->GetGovernment()->GetName()` and including System.h).

Also there are some planets that have no spaceport, but belong to a government (like Darkwaste iirc). If such knowledge could be helpful for the player (not sure, custom gov illegals maybe), I wouldn't hide the `Uninhabited` gov.

Hurleveur commented:
> Actually I think there could be a system where all planets have the same gov, but different from the system one. I know it's very unlikely, but we should support every possible situation.

does that even make sense? I mean fine I'll make it the initial government then

I wish we'd have icons for the different things and then all would be easier to stack up

Hurleveur commented:
I dont think its relevant when uninhabited systems have a government because you can always land on them anyway and they dont do anything
if anything its weird

Hurleveur committed:
look at the government

meaning that if the gov is not uninhabited it counts (instead of looking at the planet being inhabited)

TomGoodIdea commented:
> > Actually I think there could be a system where all planets have the same gov, but different from the system one. I know it's very unlikely, but we should support every possible situation.

> does that even make sense?

if a plugin added such system, it would be misleading for the player

maybe a system controlled by some gov, but all the planets occupied by another gov with strong ground defense but no spaceships, so that nobody can attack the other side? idk if that makes sense

Hurleveur commented:
well I considered the system system as the base one anyway, cant hurt
edit: also just to be clear: it's intended that I consider the display names only and not the true government

Hurleveur commented:
I thought I'd have time to get to it before going on a trip but I didnt so this will probably have to wait one more week before I can test it
That said this should work now at least (but then again things dont often go the way I expect them to)

Hurleveur committed:
made it work as it should!

I checked both with and without the gov and for the legends too, feel free to merge this now @TomGoodIdea sorry it took me so long ^^'
